.github/workflows/main.yaml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  # Run every day at 10 pm
+  #schedule:
+  #  # Temporary cron for testing.
+  #  - cron: '*/15 * * * *'
+  
+  push:
+    branches: [ "williamhe/**" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      releaseDate:
+        description: 'The date to check for new releases. Format: yyyy-mm-dd'
+        required: true
+        type: string
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      - name: Run SDK automation
+        run: time dotnet run --project build/tools/Automation/Automation.csproj
+
+        # [Important]
+        # Currently fails silently due to malformed path for Hugo.
+        # Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/runner/work/Oryx/Oryx/build\detector/__detectorNugetPackagesVersions.props'.
+        # Need to update the \ with / and check it works locally and in github actions
+        # 
+        # Since this does not impede automating dotnet releases we fail silently for now
+        # TODO: fix ./build/generateConstants.sh error in github actions workflow
+      - name: Run Generate Constants
+        run: ./build/generateConstants.sh || true
+      
+      # TODO: replace William with Oryx credentials
+      - name: Create commits
+        run: |
+          git status
+          git config user.name 'William Hernandez'
+          git config user.email 'williamhe@microsoft.com'
+          git add build/*
+          git add src/BuildScriptGenerator/*
```

Comment:
Why don't we just add all changes? I've updated generateConstants / constants.yaml so that we should be able to commit all changes from running `generateConstants.sh` without incident. I just don't know if there are usually some other changes that happen which we aren't trying to have committed? To be clear, I'm fine with only adding these 2 paths, more just curious if you noticed something

Reply:
I noticed changes in additional files when running `./build/generateConstants.sh` while having no changes to constants.yaml
![image](https://user-images.githubusercontent.com/86327553/191632418-f4eb1c6c-7c58-4c40-b3b3-649d396ac15b.png)

Reply:
I've noticed that many times sourcetree will mark a regenerated file as modified when unstaged, and then when I go to stage it then it realizes there are no updates and the file disappears. How many of those changes are meaningful?

Reply:
I've switched to add all files since there might be times when we want to add files other than the two specified folder such as  `build/`. For example when .Net8 rolls out, more files such as `build/__dotNetCoreSdkVersions.sh` and `build/dotNetCoreRunTimeVersions.sh` will automatically get updated